ts/data.ts

```diff
@@ -1,6 +1,25 @@
 const data = {
   view: 'entry-form',
-  entries: [],
+  entries: [] as FormEntry[],
   editing: null,
   nextEntryId: 1,
 };
+
+function writeData(): void {
+  const dataJSON: string = JSON.stringify(data);
+  localStorage.setItem('data-storage', dataJSON);
+}
+
+function readData(): any {
+  const dataJSON = localStorage.getItem('data-storage');
+  if (dataJSON) {
+    return JSON.parse(dataJSON);
+  } else {
+    return data;
```

Comment:
You should be returning what the object should initially be. What you were originally assigning to your `data` variable